client/src/components/charts/MapVisualisation.jsx

```diff
@@ -1,18 +1,202 @@
 import * as chart from '../../utilities/chart';
 
 require('../../../node_modules/leaflet/dist/leaflet.css');
 require('../../styles/MapVisualisation.scss');
 
+const isImage = (value) => {
+  if (typeof value === 'string' && value.match(/\.(jp(e?)g|png|gif)$/) !== null) {
+    return true;
+  }
+  return false;
+};
+
+const calculateBounds = (layerArray) => {
+  let maxLat = -90;
+  let minLat = 90;
+  let maxLong = -180;
+  let minLong = 180;
+
+  layerArray.forEach((layer) => {
+    const bounds = layer.bounds;
+    const minArray = bounds[0];
+    const maxArray = bounds[1];
+
+    minLat = Math.min(minLat, minArray[0]);
+    minLong = Math.min(minLong, minArray[1]);
+    maxLat = Math.max(maxLat, maxArray[0]);
+    maxLong = Math.max(maxLong, maxArray[1]);
+  });
+
+  return [[minLat, minLong], [maxLat, maxLong]];
+};
+
+const DisplayLayer = ({ layerData }) => {
+  const { chartValues } = layerData;
+  const radius = 2 + (parseInt(layerData.pointSize, 10) * 2);
+
+  return (
+    <div>
+      {chartValues.map((entry, index) =>
+        <CircleMarker
+          center={[entry.latitude, entry.longitude]}
+          radius={radius}
+          fillColor={entry.pointColor || '#000000'}
+          fillOpacity={1}
+          opacity={1}
+          color="#FFFFFF"
+          weight={1}
+          key={index}
+        >
+          {entry.popup.length > 0 &&
+          <Popup>
+            <ul
+              className="popupContainer"
+            >
+              {entry.popup.map((popupObject, popupIndex) =>
+                <li
+                  className="popupItem"
+                  key={popupIndex}
+                >
+                  <h4>
+                    {popupObject.title}
+                  </h4>
+                  <span>
+                    {isImage(popupObject.value) ?
+                      <a
+                        href={popupObject.value}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                      >
+                        <div className="imageContainer">
+                          <img
+                            src={popupObject.value}
+                            role="presentation"
+                          />
+                        </div>
+                      </a>
+                          :
+                      <span>
+                        {popupObject.value}
+                      </span>
+                        }
+                  </span>
+                </li>
+                )}
+            </ul>
+          </Popup>
+            }
+        </CircleMarker>
+        )
+      }
+    </div>
+  );
+};
+
+DisplayLayer.propTypes = {
+  layerData: PropTypes.object.isRequired,
+};
 
 export default function MapVisualisation({ visualisation, datasets, width, height }) {
-  const chartData = chart.getChartData(visualisation, datasets)[0];
+  let tileLayerUrl;
+  let tileLayerAttribution;
+
+  const processedLayerArray = [];
```

Comment:
I agree this would make sense as a separate function. But I think it might be better use something like `getProcessedLayerArray` rather than mapping the spec layers, because in some cases we don't want to include a layer from the spec in the display layers at all, e.g. when it has incomplete position data, or it's hidden. I'll extract a function on this basis and see how it goes